edge-hub/watchdog/linux/src/main.rs

```diff
@@ -0,0 +1,134 @@
+extern crate nix;
+
+use std::{thread};
+use std::io::Error;
+use std::process::{Command, Stdio, Child};
+use std::sync::{Arc};
+use std::sync::atomic::{AtomicBool, Ordering};
+use std::time::Duration;
+use nix::unistd::Pid;
+use nix::sys::signal::{self, Signal};
+use signal_hook::{iterator::Signals, SIGTERM};
+use tracing::{info, error};
+use tracing_subscriber;
+
+fn main() -> Result<(), Error> {
+    init_logging();
+
+    let has_received_sigterm = match register_sigterm_listener() {
+        Ok(has_received_sigterm) => has_received_sigterm,
+        Err(e) => {
+            error!("Failed to register sigterm listener. Shutting down.");
+            return Err(e)
+        }
+    };
+
+    let mut edgehub = Command::new("dotnet")
+            .arg("/app/Microsoft.Azure.Devices.Edge.Hub.Service.dll")
+            .stdout(Stdio::inherit())
+            .spawn()
+            .expect("Failed to execute Edge Hub process");
+    
+    let mut broker = Command::new("/usr/local/bin/mqttd")
```

Comment:
Not directly related to this PR, but are we should we instead put this binary in /app similar to the Edge Hub dll?

Reply:
Sure, why not. I assume it would be just Dockerfile/script change?

Reply:
yep